mmcv/runner/hooks/optimizer.py

```diff
@@ -276,3 +290,7 @@ def after_train_iter(self, runner):
             if has_overflow:
                 runner.logger.warning('Check overflow, downscale loss scale '
                                       f'to {self.loss_scaler.cur_scale}')
+
+            # save state_dict of loss_scaler
+            runner.meta.setdefault(
```

Comment:
Shall we reduce the time to update self.loss_scalar? For example, only update the meta when `has_overflow` and at the initialization and resuming.

Reply:
I agree that we shall reduce the frequency if possible.

For mmcv based fp16_optimizer, we can do this by checking the value of `has_overflow`. 
But for pytorch.amp based fp16_optimizer, we have to hack the internal status of the loss scaler for `has_overflow`. 
In order to uniform the codes and preserve better readability, the current solution looks better to me. 

Besides, compare with some `if condition`, updating a small and simple dict does not too much extra cost.